### PR TITLE
lab02: kernel_modules: Add extra exercises

### PR DESCRIPTION
**Pull request by andanicolae**
Signed-off-by: Anda Nicolae <nicolae.anda@gmail.com>

### CODE REVIEW

File: Documentation/teaching/labs/kernel_modules.rst [hunk 1]
```diff
@@ -973,7 +973,7 @@ module.
 .. note::
           * In the Linux kernel, a process is described by the
 	    :c:type:`struct task_struct`.  Use |LXR|_ to find the
-	    definition of ``struct task_struct``.
+            definition of :code:`struct task_struct`.
```
Comment (tavip) [inline Documentation/teaching/labs/kernel_modules.rst]: Use :c:type:`struct task_struct` to link to kernel-docs definitions.

File: Documentation/teaching/labs/kernel_modules.rst [hunk 2]
```diff
@@ -994,3 +994,112 @@ displayed processes differ. This is because a module is being loaded
...
+
+Analyze the stacktrace and determine the code that generated the bug.
+How can we find out from KDB the address where the module was loaded?
+
+In parallel, use GDB in a new window to view the code based on KDB information.
+
+.. hint::
+    Load the symbol file. Use :command:`info line`.
+
+When writing to :file:`/proc/hello_kdb_break`, the module will increment the :code:`kdb_write_address` variable. Enter KDB and set a breakpoint for each write access of the :code:`kdb_write_address` variable. Return to kernel to trigger a write using:
```
Comment (tavip) [inline Documentation/teaching/labs/kernel_modules.rst]: Please use a maximum of 79 chars per line.

File: Documentation/teaching/labs/kernel_modules.rst [hunk 3]
```diff
@@ -994,3 +994,148 @@ displayed processes differ. This is because a module is being loaded
...
+
+Create a kernel module that displays the virtual memory areas of the current
+process; for each memory area it will display the start address and the end
+address.
+
+.. hint::
+    * Start from an existing kernel module.
+
+    * Investigate the structures
+      `struct task_struct <https://elixir.bootlin.com/linux/v4.15.7/source/include/linux/sched.h#L520>`_,
```
Comment (tavip) [inline Documentation/teaching/labs/kernel_modules.rst]: Please don't use LXR links, use :c:type:`struct task_struct` so we can integrate with kernel docs.

**Comment by tavip** [comment]
@andanicolae I think these changes got merged via another PR, right? If so, please close this PR.

**Comment by tavip** [comment]
@andanicolae I've removed the LXR links and done some more 80 cols fixes, remove some end of line whitespace, please take a look and let me know if there are ok.

File: Documentation/teaching/labs/kernel_modules.rst [hunk 4]
```diff
@@ -143,40 +145,45 @@ For the example above, the steps to compile are:
...
+
+   * any other target suffix will be ignored by :file:`Kbuild` and will not be
+     compiled
+
+
+.. note:: These suffixes are used to easily configure the kernel by running the
+          :command:`make menuconfig` command or directly editing the
+          :file:`.config` file. This file sets a series of variables that are
+          used to determine which features are added to the kernel at build
+          time. For example, when adding BTRFS support with :command: `make
```
Comment (andanicolae) [inline Documentation/teaching/labs/kernel_modules.rst]: :command:`make menuconfig` no space after :

File: Documentation/teaching/labs/kernel_modules.rst [hunk 5]
```diff
@@ -143,40 +145,45 @@ For the example above, the steps to compile are:
...
+          menuconfig`, add the line :code:`CONFIG_BTRFS_FS = y` to the
+          :file:`.config` file.  The BTRFS kbuild contains the line
+          ``obj-$(CONFIG_BTRFS_FS):= btrfs.o``, which becomes ``obj-y:=
+          btrfs.o``. This will compile the :file:`btrfs.o` object and will be
+          linked to the kernel. Before the variable was set, the line became
+          ``obj:=btrfs.o`` and so it was ignored, and the kernel was build-at
```
Comment (andanicolae) [inline Documentation/teaching/labs/kernel_modules.rst]: was builded

File: Documentation/teaching/labs/kernel_modules.rst [hunk 6]
```diff
@@ -187,10 +194,10 @@ the routine specified as a parameter of the ``module_exit`` will be executed.
 A complete example of compiling and loading/unloading mode is presented below:
```
Comment (andanicolae) [inline Documentation/teaching/labs/kernel_modules.rst]: loading/unloading a kernel module

File: Documentation/teaching/labs/kernel_modules.rst [hunk 7]
```diff
@@ -408,14 +417,13 @@ objdump
 -------
 
 Detailed information about the instruction that generated the oops can be found
-using the :command:`objdump` utility. Useful options to use are :command:`-d` to
-disassemble the code and :command:`-S`` for interleaving C code in assembly language
-code.
-For efficient decoding, however, we need the address where the kernel module was
-loaded. This can be found in :file:`/proc/modules`.
+using the :command:`objdump` utility. Useful options to use are :command:`-d`
+to disassemble the code and :command:`-S`` for interleaving C code in assembly
```
Comment (andanicolae) [inline Documentation/teaching/labs/kernel_modules.rst]: :command:`-S` (remove back quote from the end)

File: Documentation/teaching/labs/kernel_modules.rst [hunk 8]
```diff
@@ -1040,66 +1064,96 @@ When writing to :file:`/proc/hello_kdb_break`, the module will increment the :co
 2. PS Module
 ------------
 
-Update the created kernel module at :ref:`proc-info` in order to display information about all the processes from the system, when inserting the kernel module, not just about the current process. Afterwards, compare the obtained result with the output of the :command:`ps` command.
+Update the created kernel module at :ref:`proc-info` in order to display
+information about all the processes from the system, when inserting the kernel
```
Comment (andanicolae) [inline Documentation/teaching/labs/kernel_modules.rst]: processes in the system, my bad

**Pull request merged**